pkg/api/file_repository.go

```diff
@@ -20,28 +20,39 @@ type FileSystemRepository struct {
 // NewFileSystemRepository creates a mongo country repo
 func NewFileSystemRepository(dir string) (*FileSystemRepository, error) {
 	repo := &FileSystemRepository{definitions: make(map[string]*Definition)}
+
 	// Grab json files from directory
 	files, err := ioutil.ReadDir(dir)
 	if nil != err {
 		return nil, err
 	}
 
 	for _, f := range files {
-		if strings.Contains(f.Name(), ".json") {
-			filePath := filepath.Join(dir, f.Name())
-			log.WithField("path", filePath).Info("Loading API definition from file")
-			appConfigBody, err := ioutil.ReadFile(filePath)
-			if err != nil {
-				log.WithError(err).WithField("path", filePath).Error("Couldn't load the api definition file")
-				return nil, err
+		filePath := filepath.Join(dir, f.Name())
+		definition := new(Definition)
+
+		v := viper.New()
+		v.SetConfigFile(filePath)
+		v.WatchConfig()
+		v.OnConfigChange(func(e fsnotify.Event) {
+			log.Debug("API configuration changed, reloading...")
+			if err := v.Unmarshal(definition); err != nil {
```

Comment:
`e.Op` can be `Remove`, then no need to unmarshal, or `Rename`, then I do not know if `viper` handles this properly and re-reads new file. I think you should leave only logging part for now and implement this in `hot-reload` feature.